s3/s3.go

```diff
@@ -206,16 +208,16 @@ func (c *S3CliClient) UploadFile(logger lager.Logger, client *s3.Client, localFi
 
 	logger.Info(fmt.Sprintf("S3 putting local file: %s into bucket %s with remote file: %s ", localFilePath, bucketName, remotePath))
 
-	file, err := os.Open(localFilePath)
+	readFile, err := os.ReadFile(localFilePath)
 	if err != nil {
-		return err
+		return fmt.Errorf("UploadFile: failed to read local file path: %v", err)
 	}
-	defer file.Close()
-
-	_, err = client.PutObject(context.TODO(), &s3.PutObjectInput{
+	largeBuffer := bytes.NewReader(readFile)
```

Comment:
NIT: it's just a buffer, the size is determined by the `uploader.Upload` function